cli/cmd/completion.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+const completionDesc = `
+Generate autocompletion script for lokoctl for the specified shell (bash or zsh).
+`
+
+const completionExample = `  # Load the lokoctl completion code for bash into the current shell.
+  source <(lokoctl completion bash)
+
+  # Load the lokoctl completion code for zsh into the current shell.
+  source <(lokoctl completion zsh)
+
+  # Generate Bash completion file and load it for every shell.
+  lokoctl completion bash > ~/.bash_lokoctl_completion
+  echo "source ~/.bash_lokoctl_completion" >> ~/.bashrc && source ~/.bashrc
+
+  # Set the lokoctl completion code for zsh to autoload on startup.
+  lokoctl completion zsh > "${fpath[1]}/_lokoctl" && exec $SHELL`
+
+const bashCompDesc = `  ### If running Bash 3.2 that is included with macOS, install bash completion using homebrew.
+  brew install bash-completion
+	
+  ### If running Bash 4.1+ on macOS, install bash completion using homebrew.
+  brew install bash-completion@2
+
+  ### Load the lokoctl completion code for bash into the current shell.
+  source <(lokoctl completion bash)
+
+  ### Generate Bash completion file and load it for every shell.
+  lokoctl completion bash > ~/.bash_lokoctl_completion
+  echo "source ~/.bash_lokoctl_completion" >> ~/.bashrc && source ~/.bashrc
+`
+
+const zshCompDesc = `  Generate the autocompletion script for lokoctl for the zsh shell.
+
+  ### Load the lokoctl completion code for zsh into the current shell.
```

Comment:
We seem to be using "autocompletion" and "completion" randomly. We should settle on one and use it consistently. Same for "code" vs "script" (I prefer "code").